relay-event-normalization/src/normalize/mod.rs

```diff
@@ -288,27 +288,34 @@ impl ModelCosts {
     }
 }
 
-/// Regex that matches version or date patterns at the end of a model name
-///
-/// Pattern breakdown (matches either):
-/// 1. Version pattern: [-_]v\d+[:\.]?\d*([:\-].*)?
-///    - Separator: [-_]
-///    - Version: v\d+[:\.]?\d* (must start with 'v': v1, v1.0, v1:0)
-///    - Optional trailing content: [:\-].*
-/// 2. Date pattern: ([-_@])?(\d{4}[-/\.]\d{2}[-/\.]\d{2}|\d{8})
-///    - Optional separator: [-_@]?
-///    - YYYY-MM-DD with -, /, or . OR YYYYMMDD
+/// Regex that matches version and/or date patterns at the end of a model name
 ///
 /// Examples matched:
-/// - "-v1:0" in "claude-sonnet-4-20250514-v1:0"
-/// - "_v2" in "model_v2"
-/// - "-v1.0" in "gpt-4-v1.0"
-/// - "-20250522" in "claude-4-sonnet-20250522"
-/// - "-2025-06-10" in "o3-pro-2025-06-10"
-/// - "@20241022" in "claude-3-5-haiku@20241022"
+/// - "-20250805-v1:0" in "claude-opus-4-1-20250805-v1:0" (both)
+/// - "-20250514-v1:0" in "claude-sonnet-4-20250514-v1:0" (both)
+/// - "_v2" in "model_v2" (version only)
+/// - "-v1.0" in "gpt-4-v1.0" (version only)
+/// - "-20250522" in "claude-4-sonnet-20250522" (date only)
+/// - "-2025-06-10" in "o3-pro-2025-06-10" (date only)
+/// - "@20241022" in "claude-3-5-haiku@20241022" (date only)
 static VERSION_OR_DATE_REGEX: LazyLock<Regex> = LazyLock::new(|| {
```

Comment:
```suggestion
static VERSION_AND_DATE_REGEX: LazyLock<Regex> = LazyLock::new(|| {
```

Feels like this now matches both not necessarily either, but got no strong opinion.